tests/virt_autotest/restore_guests.pm

```diff
@@ -0,0 +1,32 @@
+# Summary: make the guests(specified by test suite settings) ready(virsh define).
+# Maintainer: Julie CAO <jcao@suse.com>
+package restore_guests;
+
+use strict;
+use warnings;
+use testapi;
+use base "virt_autotest_base";
+use virt_utils qw(get_guest_list remove_vm restore_downloaded_guests);
+
+sub run {
+    my $guest_list         = get_guest_list();
+    my $downloaded_xml_dir = "/tmp/download_vm_xml";
+
+    #clean up env
+    my $listed_guests = script_output "virsh list --all | sed -n '/^-/,\$p' | sed '1d;/Domain-0/d' | awk '{print \$2;}'";
```

Comment:
I suggest use "virsh list --all --name" to avoid such lengthy query.

Reply:
Yes, it's much shorter :+1: